consensus/src/metrics_safety_rules.rs

```diff
@@ -125,3 +148,121 @@ impl TSafetyRules for MetricsSafetyRules {
         })
     }
 }
+
+pub struct MockSafetyRules {
+    // number of initialize() calls
+    init_calls: i32,
+
+    // max initialize() calls to complete perform_initialize()
+    max_init_calls: i32,
+}
+
+impl MockSafetyRules {
+    #[allow(dead_code)]
+    pub fn new(init_calls: i32, max_init_calls: i32) -> Self {
+        Self {
+            init_calls,
+            max_init_calls,
+        }
+    }
+}
+
+impl TSafetyRules for MockSafetyRules {
+    fn consensus_state(&mut self) -> Result<ConsensusState, Error> {
```

Comment:
nits: I don't think we need all of these comments in this mock implementation. This is only test code and almost all values are defaults (e.g., 0, None, etc.).

Reply:
Removed all unnecessary comments.